packages/router/src/Exceptions/HttpErrorResponseProcessor.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Tempest\Router\Exceptions;
+
+use Tempest\Core\AppConfig;
+use Tempest\Http\HttpException;
+use Tempest\Http\Response;
+use Tempest\Router\ResponseProcessor;
+
+final readonly class HttpErrorResponseProcessor implements ResponseProcessor
+{
+    public function __construct(
+        private AppConfig $appConfig,
+    ) {}
+
+    public function process(Response $response): Response
+    {
+        // Throwing an HttpException during tests would make testing more
+        // complex, and is not strictly needed. During development, we
+        // don't need exceptions either, since the exception handler
+        // is different. For this reason, we skip processing here.
+        if (! $this->appConfig->environment->isProduction()) {
```

Comment:
Shouldn't we make a difference here between exceptions that were thrown on the framework level, and user-based exceptions? There are a lot of cases where I want tests to simply throw the exception instead of hiding it behind a response. 

It could also be a toggle. This is what we used to do in tests: `$this->http->throwExceptions()->get(…)`

WDYT?

Reply:
Conceptually I agree yes. Not sure how to implement though

Reply:
This actually still works, so it's fine for tests. I don't think we need to followup for now. Let's wait and see if people require more control based on them playing around with Tempest